libs/dh/metering-point/feature-search/src/lib/dh-search.component.ts

```diff
@@ -73,28 +77,56 @@ import { dhMeteringPointIdValidator } from './dh-metering-point.validator';
           }
         </watt-text-field>
       </div>
+
+      @if (mpNotFound()) {
+        <watt-empty-state size="small" icon="custom-no-results" [title]="t('noResultFound')" />
+      }
     </vater-stack>
   `,
 })
 export class DhSearchComponent {
   private router = inject(Router);
+  private activatedRoute = inject(ActivatedRoute);
   private doesMeteringPointExist = lazyQuery(DoesMeteringPointExistDocument);
 
-  searchControl = new FormControl('', [dhMeteringPointIdValidator()]);
+  searchControl = new FormControl('', {
+    validators: [dhMeteringPointIdValidator()],
+    nonNullable: true,
+  });
+
+  mpNotFound = signal(false);
+
+  constructor() {
+    this.activatedRoute.queryParamMap.pipe(takeUntilDestroyed()).subscribe((params) => {
+      const maybeMeteringPointId = params.get(dhMeteringPointIdParam);
+
+      if (maybeMeteringPointId) {
+        this.mpNotFound.set(true);
+        this.searchControl.setValue(maybeMeteringPointId);
+      } else {
+        this.mpNotFound.set(false);
+        this.searchControl.reset();
+      }
+    });
+
+    this.searchControl.valueChanges
+      .pipe(distinctUntilChanged(), takeUntilDestroyed())
+      .subscribe(() => this.mpNotFound.set(false));
+  }
 
   async onSubmit() {
     this.searchControl.markAsTouched();
 
     if (this.searchControl.invalid) return;
 
     const meteringPointId = this.searchControl.getRawValue();
-
-    if (!meteringPointId) return;
-
     const result = await this.doesMeteringPointExist.query({ variables: { meteringPointId } });
 
-    if (!result.data?.meteringPoint.meteringPointId) return;
+    if (!result.data) {
+      this.mpNotFound.set(true);
```

Comment:
Suggestion: use computed to compute if the meteringpoint was found or not

Reply:
This is a bit tricky because there are a multiple cases where `mpNotFound` can be set. I couldn't figure it out a good solution. Have kept it the same for now. Need to experiment some more. Also welcome to suggestions.